docs/markdown/snippets/cargo_features.md

```diff
@@ -0,0 +1,14 @@
+## Cargo features are resolved globally
+
+When configuring a Cargo dependency, Meson will now resolve its complete
+dependency tree and feature set before generating the subproject AST.
+This solves many cases of Cargo subprojects being configured with missing
+features that the main project had to enable by hand using e.g.
+`default_options: ['foo-rs:feature-default=true']`.
+
+Note that there could still be issues in the case there are multiple Cargo
+entry points. That happens if the main Meson project makes multiple `dependency()`
+calls for different Cargo crates that have common dependencies.
+
+Breaks: This change removes per feature Meson option That were previously
+possible to set as show above or from command line `-Dfoo-rs:feature-foo=true`.
```

Comment:
Wouldn't this also break projects that are setting those options via `default_options: [...]`?

Reply:
Yes that's an incompatible change for cargo subprojects, but we never claimed that was stable. Hopefully not many projects actually uses it yet.

Reply:
I'm looking and I cannot find a single place where we claim it is *not* stable, either as a meson warning (the way we print for `import()` of an unstable module) or in the documentation.

I think the technical term for this is "we blew it".

Reply:
It's here: https://github.com/mesonbuild/meson/pull/12358. I asked many times to merge that before release...

Reply:
> I think the technical term for this is "we blew it".

yes... :/

Reply:
Actually my PR for that was https://github.com/mesonbuild/meson/pull/12356 but then @dcbaker wanted to make a more complex but better solution, and we lost track if it... :/

Reply:
Yeah, we did. I meant to just merge 12356 and replace it later if I ever got 12358 working. That's on me